cpp/include/cudf/detail/gather.cuh

```diff
@@ -116,24 +116,38 @@ struct column_gatherer_impl
    * @param source_column View into the column to gather from
    * @param gather_map_begin Beginning of iterator range of integral values representing the gather map
    * @param gather_map_end End of iterator range of integral values representing the gather map
+   * @param policy Mask allocation policy for creation of gathered column
    * @param ignore_out_of_bounds Ignore values in `gather_map` that are out of bounds
    * @param mr Memory resource to use for all allocations
    * @param stream CUDA stream on which to execute kernels
    *---------------------------------------------------------------------------**/
     std::unique_ptr<column> operator()(column_view const& source_column,
                                        MapIterator gather_map_begin,
                                        MapIterator gather_map_end,
+                                       cudf::experimental::mask_allocation_policy policy,
```

Comment:
This doesn't make sense to me. Shouldn't the state of the output null mask depend only on the existence of nulls in the source column?

Reply:
Okay, I understand. If `ignore_out_of_bounds` is true, you need to have a null mask allocated regardless of the presence of nulls in the source. The `mask_allocation_policy` parameter feels very awkward in this API. Instead, why not just always allocate a bitmask for the output column if `ignore_out_of_bounds == true`?

Reply:
That would definitely make it easier. However, my motivation was to avoid allocation of masks if all bits are supposed to be valid.

Reply:
I'd prefer having an unnecessary allocation to adding a `policy` argument to the `gather` API. Otherwise it introduces possibility for user error like setting `policy == NEVER` when `ignore_out_of_bounds == true` and there _are_ out of bound indices.

Reply:
Ok that's a lot easier. Will fix it